node.set has been deprecated in favor of using explicit precedence level

iamthemuffinman opened the pull request:
node.set will be removed in Chef 14:

https://docs.chef.io/deprecations_attributes.html#set-and-set-unless

shortdudey123 commented on recipes/mongodb_org_repo.rb:
```diff
@@ -47,7 +47,7 @@
     key package_version_major >= 3.2 ? 'EA312927' : '7F0CEB10'
   end
-when 'amazon', 'fedora', 'oracle', 'rhel'
+when 'rhel'
```
Previous line was correct, revert please

iamthemuffinman replied:
Gah, I was testing with Foodcritic here and forgot to change this line. Foodcritic complains about the way it was originally but behaves normally when 'oracle' is removed because the 'oracle' platform is included when using the 'rhel' platform family since at least Ohai 7.2.0 (https://github.com/chef/ohai/blob/7.2.0/lib/ohai/plugins/linux/platform.rb#L112).

shortdudey123 replied:
Ah missed that, remove only `oracle` then :)
`amazon` and `fedora` are families

shortdudey123 commented on test/unit/default_spec.rb:
```diff
@@ -17,20 +17,20 @@
 
   it 'should disable logpath when syslog is set' do
-    chef_run.node.set.mongodb.config.syslog = true
+    chef_run.node.default.mongodb.config.syslog = true
```
method access is deprecated too

iamthemuffinman replied:
Definitely spaced this. Thanks!

shortdudey123 commented on resources/user.rb:
```diff
@@ -8,7 +8,4 @@
-  super
-  @action = :add
-end
+default_action :add
```
This should go under the `actions` directive

iamthemuffinman replied:
Foodcritic also complained about this one. It seems the default action is implicitly the first one specified which in this case would be 'add': https://docs.chef.io/custom_resources.html#default-action

I would prefer it explicitly defined, but ultimately it's up to you.

shortdudey123 commented on recipes/replicaset.rb:
```diff
@@ -19,8 +19,8 @@
-node.set['mongodb']['is_replicaset'] = true
-node.set['mongodb']['cluster_name'] = node['mongodb']['cluster_name']
+node.default['mongodb']['is_replicaset'] = true
+node.default['mongodb']['cluster_name'] = node['mongodb']['cluster_name']
```
This chunk should be `node.normal`

shortdudey123 commented on recipes/shard.rb:
```diff
@@ -21,10 +21,10 @@
-node.set['mongodb']['shard_name'] = node['mongodb']['shard_name']
-node.set['mongodb']['is_replicaset'] = node['mongodb']['is_replicaset']
-node.set['mongodb']['cluster_name'] = node['mongodb']['cluster_name']
+node.default['mongodb']['is_shard'] = true
```
This chunk should be `node.normal`

shortdudey123 commented on Rakefile:
```diff
@@ -35,7 +35,10 @@ begin
 
   desc 'Run ChefSpec examples'
-  RSpec::Core::RakeTask.new(:spec)
+  RSpec::Core::RakeTask.new(:spec) do |t|
```
revert this, i will rip the spec apart once the rest of the rewrite is done
Unless you are volunteering :)

shortdudey123 commented on recipes/mongodb_org_repo.rb:
```diff
@@ -47,7 +47,7 @@
     key package_version_major >= 3.2 ? 'EA312927' : '7F0CEB10'
   end
-when 'amazon', 'fedora', 'oracle', 'rhel'
+when 'rhel', 'fedora', 'amazon'
```
put this back in alphabetical order please

iamthemuffinman commented:
@shortdudey123 Should be good now. Also, I modified the unit tests just a little bit so that they would pass if ran manually.

shortdudey123 commented on test/unit/default_spec.rb:
```diff
@@ -4,34 +4,27 @@
   it 'should be able to set logpath to nil, and wont create' do
-    chef_run.node.set.mongodb.config.logpath = '/tmp/mongodb_config_logpath/logfile.log'
-    chef_run.node.set.mongodb.config.logpath = nil
+    chef_run.node.normal['mongodb']['config']['logpath'] = '/tmp/mongodb_config_logpath/logfile.log'
```
node['mongodb']['config']['logpath'] does not exist

shortdudey123 commented on test/unit/install_spec.rb:
```diff
@@ -4,24 +4,24 @@
 
   it 'package install mongodb-org via mongodb-org' do
-    chef_run.node.set.mongodb.install_method = 'mongodb-org'
+    chef_run.node.normal['mongodb']['install_method'] = 'mongodb-org'
```
this is the default so not needed

shortdudey123 commented:
Can you move all of the spec stuff into a separate PR since it is unrelated to the original change

shortdudey123 commented on definitions/mongodb.rb:
```diff
@@ -175,7 +174,6 @@
     mode '0755'
     action :create
     recursive true
-    not_if { new_resource.is_mongos }
```
revert

iamthemuffinman commented:
@shortdudey123 I got a little carried away with the changes and reverted everything not related to this PR. I also squashed all those commits as to not make a bunch of noise in the commit tree. Let me know if there's anything else.

shortdudey123 commented:
No problem and thanks!
The spec tests are going to take quite a bit to fixup due to how much i changed :)
